yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -95,6 +95,10 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
       throw new IllegalArgumentException(
         "You must specify at least 1 executor!\n" + getUsageMessage())
     }
+    if (executorCores < sparkConf.getInt("spark.task.cpus", 1)) {
+      throw new SparkException("spark.executor.cores must not be less than " +
```

Comment:
only nit here is that I might not have specified it via spark.executor.cores but rather via the spark-submit --executor-cores option.  Perhaps we should just say "Executor cores must".